### PR TITLE
reimbursement builder now building correctly with multiple grants

### PR DESCRIPTION
**Pull request by sbillinge**
This fixes a couple of bugs in the original.  It is also now extended to allow more than one grant to be billed, and it takes the grant value directly from the grant field an not from the project field.  This is a feature in the current case, because I want to set the grant myself and not have it change later if a project updates its grant informtion.  I still need to make it PEP8 complient and make sure it passes tests, but it is nearly ready for review.

### CODE REVIEW

**Comment by CJ-Wright** [comment]
I would use black for the formatting (https://github.com/python/black)

**Comment by sbillinge** [comment]
Not sure why this is failing also....subprocessError, but why?

**Comment by CJ-Wright** [comment]
This is the real error
```python
Traceback (most recent call last):
  File "/home/travis/miniconda/envs/test/bin/regolith", line 4, in <module>
    __import__('pkg_resources').run_script('regolith==0.4.0', 'regolith')
  File "/home/travis/miniconda/envs/test/lib/python3.7/site-packages/pkg_resources/__init__.py", line 666, in run_script
    self.require(requires)[0].run_script(script_name, ns)
  File "/home/travis/miniconda/envs/test/lib/python3.7/site-packages/pkg_resources/__init__.py", line 1453, in run_script
    exec(code, namespace, namespace)
  File "/home/travis/miniconda/envs/test/lib/python3.7/site-packages/regolith-0.4.0-py3.7.egg/EGG-INFO/scripts/regolith", line 3, in <module>
    main()
  File "/home/travis/miniconda/envs/test/lib/python3.7/site-packages/regolith-0.4.0-py3.7.egg/regolith/main.py", line 283, in main
    CONNECTED_COMMANDS[rc.cmd](rc)
  File "/home/travis/miniconda/envs/test/lib/python3.7/site-packages/regolith-0.4.0-py3.7.egg/regolith/commands.py", line 113, in build
    bldr.build()
  File "/home/travis/miniconda/envs/test/lib/python3.7/site-packages/regolith-0.4.0-py3.7.egg/regolith/builders/basebuilder.py", line 100, in build
    getattr(self, cmd)()
  File "/home/travis/miniconda/envs/test/lib/python3.7/site-packages/regolith-0.4.0-py3.7.egg/regolith/builders/reimbursementbuilder.py", line 64, in excel
    float(percent) / 100.0 for percent in ex["grant_percentages"]
  File "/home/travis/miniconda/envs/test/lib/python3.7/site-packages/regolith-0.4.0-py3.7.egg/regolith/chained_db.py", line 50, in __getitem__
    raise KeyError("{} is none of the current mappings".format(key))
KeyError: 'grant_percentages is none of the current mappings'
```

**Comment by sbillinge** [comment]
@scopatz this should be ready now.  Thanks

File: regolith/builders/reimbursementbuilder.py [hunk 1]
```diff
@@ -60,78 +54,96 @@ def construct_global_ctx(self):
...
-                    row=r,
-                    column=ue_column,
-                    value=item.get("unsegregated_expense", 0),
+                grants = [
+                    fuzzy_retrieval(gtx["grants"], ["alias", "name", "_id"], grant)
+                    for grant in ex["grants"]
+                ]
+                #            grants = [doc for doc in gtx["grants"] if
+                #                     doc.get("alias") == project["grant"]]
+                #            grant = grants[0]
```
Comment (scopatz) [inline regolith/builders/reimbursementbuilder.py]: Please remove dead code

**Comment by scopatz** [comment]
one small change

**Comment by sbillinge** [comment]
Thanks @scopatz I cleaned the dead code.

**Comment by scopatz** [comment]
Thanks!

**Comment by sbillinge** [comment]
Thanks!

**Pull request merged**